File: legacy/pasarguardbot/frontend/src/types/panel/texts.ts
/** Mirrors app/models/panel/texts.py */
import type { PanelAuthRequest, PanelEnvelope } from "./common";

export const BANNER_POSITIONS = ["", "top", "bottom"] as const;

/** One editable key: its definition plus the stored override, if any. */
export interface PanelTextEntry {
  key: string;
  title?: string | null;
  placeholders: Record<string, string>;
  value?: string | null;
  lang?: string | null;
  banner_url?: string | null;
  banner_position?: string | null;
  stored: boolean;
}

export interface PanelTextSection {
  key: string;
  name?: string | null;
  icon?: string | null;
  entries: PanelTextEntry[];
}

export interface PanelTextsRequest extends PanelAuthRequest {
  q?: string;
}

export interface PanelTextsResponse extends PanelEnvelope {
  sections: PanelTextSection[];
  banner_positions: string[];
}

export interface PanelTextSaveRequest extends PanelAuthRequest {
  key: string;
  value?: string;
  lang?: string;
  banner_url?: string;
  banner_position?: string;
}

export interface PanelTextDeleteRequest extends PanelAuthRequest {
  key: string;
  lang?: string;
}

File: legacy/pasarguardbot/app/models/panel/texts.py
"""Admin panel DTOs: the editable bot texts."""

from pydantic import BaseModel, Field

from app.models.panel.common import PanelRequest, PanelResponse

BANNER_POSITIONS = ("", "top", "bottom")


class PanelTextEntry(BaseModel):
    """One editable key: its definition plus the stored override, if any."""

    key: str
    title: str | None = None
    placeholders: dict[str, str] = Field(default_factory=dict)
    value: str | None = None
    lang: str | None = None
    banner_url: str | None = None
    banner_position: str | None = None
    stored: bool = False


class PanelTextSection(BaseModel):
    key: str
    name: str | None = None
    icon: str | None = None
    entries: list[PanelTextEntry] = Field(default_factory=list)


class PanelTextsRequest(PanelRequest):
    q: str = Field("", max_length=64)


class PanelTextsResponse(PanelResponse):
    sections: list[PanelTextSection] = Field(default_factory=list)
    banner_positions: list[str] = Field(default_factory=lambda: list(BANNER_POSITIONS))


class PanelTextSaveRequest(PanelRequest):
    key: str = Field(..., min_length=1, max_length=100)
    value: str = Field("", max_length=8192)
    lang: str = Field("", max_length=10)
    banner_url: str = Field("", max_length=500)
    banner_position: str = Field("", max_length=10)


class PanelTextDeleteRequest(PanelRequest):
    key: str = Field(..., min_length=1, max_length=100)
    lang: str = Field("", max_length=10)
